### ISSUE TITLE
/api/authenticate/baseauth expects email address instead of username

### ISSUE THREAD

**Issue by innovate-invent** [galaxyproject/galaxy#10519]
`POST /api/users` and then `GET /api/authenticate/baseauth` returns
```json
{"err_msg": "The user does not exist.", "err_code": 404001}
```
a second `POST /api/users` returns 400008: User with that email already exists.

```
172.17.0.1 - - [21/Oct/2020:23:31:54 +0000] "POST /api/users HTTP/1.1" 200 - "-" "terraform-provider-galaxy/galaxy - blend4go"
[pid: 765|app: 0|req: 1/1] 172.17.0.1 () {40 vars in 535 bytes} [Wed Oct 21 23:31:53 2020] POST /api/users => generated 247 bytes in 94 msecs (HTTP/1.1 200) 3 headers in 139 bytes (1 switches on core 2)
172.17.0.1 - - [21/Oct/2020:23:31:57 +0000] "GET /api/authenticate/baseauth HTTP/1.1" 404 - "-" "terraform-provider-galaxy/galaxy - blend4go"
[pid: 765|app: 0|req: 2/2] 172.17.0.1 () {42 vars in 611 bytes} [Wed Oct 21 23:31:57 2020] GET /api/authenticate/baseauth => generated 59 bytes in 17 msecs (HTTP/1.1 404) 3 headers in 146 bytes (1 switches on core 0)
172.17.0.1 - - [21/Oct/2020:23:33:40 +0000] "POST /api/users HTTP/1.1" 400 - "-" "terraform-provider-galaxy/galaxy - blend4go"
[pid: 765|app: 0|req: 3/3] 172.17.0.1 () {40 vars in 535 bytes} [Wed Oct 21 23:33:40 2020] POST /api/users => generated 119 bytes in 14 msecs (HTTP/1.1 400) 3 headers in 148 bytes (1 switches on core 3)
172.17.0.1 - - [21/Oct/2020:23:33:40 +0000] "GET /api/users?deleted=true&f_name=test HTTP/1.1" 200 - "-" "terraform-provider-galaxy/galaxy - blend4go"
[pid: 765|app: 0|req: 4/4] 172.17.0.1 () {38 vars in 556 bytes} [Wed Oct 21 23:33:40 2020] GET /api/users?deleted=true&f_name=test => generated 2 bytes in 11 msecs (HTTP/1.1 200) 3 headers in 139 bytes (1 switches on core 1)
```

I confirmed that both requests were using the same username/password.

`GET /api/authenticate/baseauth` request headers:
X-Api-Key admin
Authorization: Basic YmFzaWMxMDpwYXNzd29yZA==
Accept application/json

`GET http://localhost:8080/api/users/c9468fdb6dc5c5f1/api_key/inputs`
```json
{"inputs": [{"help": " An API key will allow you to access via web API. Please note that this key acts as an alternate means to access your account and should be treated with the same care as your login password.", "value": "Not available.", "label": "Current API key:", "readonly": true, "type": "text", "name": "api-key"}], "message": ""}
```

`POST http://localhost:8080/api/users/c9468fdb6dc5c5f1/api_key`
`GET http://localhost:8080/api/users/c9468fdb6dc5c5f1/api_key/inputs`
```json
{"inputs": [{"help": " An API key will allow you to access via web API. Please note that this key acts as an alternate means to access your account and should be treated with the same care as your login password.", "value": "3e00f155ac0eb9d77dba0a26e0aca5c6", "label": "Current API key:", "readonly": true, "type": "text", "name": "api-key"}], "message": ""}
```

`GET /api/authenticate/baseauth` continues to fail.

I tries unsetting X-Api-Key in the header just in case. I know this worked in previous Galaxy versions, my application depends on it.
Galaxy 20.09

**Comment by dannon** [comment]
(following up on gitter, but for posterity -- this seems like it's working to me)
```
[yoplait@incus:~/work/galaxy] curl --location --request POST '192.168.0.101:8080/api/users' \
--header 'x-api-key: 824f876b879557fec51189193abd280e' \
--header 'Content-Type: application/json' \
--data-raw '{
    "username": "foobizbar2",
    "email": "foobizbar2@blah.com",
    "password": "password"
}' && curl --user "foobizbar2@blah.com:password" http://192.168.0.101:8080/api/authenticate/baseauth
{"model_class": "User", "id": "0c5ffef6d88a1e97", "email": "foobizbar2@blah.com", "username": "foobizbar2", "total_disk_usage": 0.0, "nice_total_disk_usage": "0 bytes", "deleted": false, "active": true, "last_password_change": "2020-10-21T22:05:45.391731"}{"api_key": "babdd9aa9a3a69c990133155b360bce6"}%
```

**Comment by innovate-invent** [comment]
OK, so the issue is that it wants the email address/password rather than the expected username/password.

This really should be changed.

**Comment by innovate-invent** [comment]
Closed by #10521
